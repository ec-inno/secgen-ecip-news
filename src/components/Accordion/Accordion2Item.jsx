/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from '../Icon';

const Accordion2Item = ({ id, toggle, onClick, level, children }) => {
  const HeadingTag = `h${level || 3}`;

  return (
    <div className="ecl-accordion2__item">
      {!!(toggle && toggle.label) && (
        <HeadingTag className="ecl-accordion2__title">
          <button
            type="button"
            className={classnames(toggle.className, 'ecl-accordion2__toggle')}
            data-ecl-accordion2-toggle
            aria-controls={`${id}-content`}
            onClick={onClick}
          >
            <span className="ecl-accordion2__toggle-flex">
              <Icon
                className="ecl-accordion2__toggle-icon"
                shape={toggle.iconShape}
                size="m"
                data-ecl-accordion2-icon
              />
              <span className="ecl-accordion2__toggle-title">
                {toggle.label}
              </span>
            </span>
          </button>
        </HeadingTag>
      )}

      <div
        className="ecl-accordion2__content"
        id={`${id}-content`}
        hidden
        role="region"
      >
        {children}
      </div>
    </div>
  );
};

Accordion2Item.propTypes = {
  id: PropTypes.string,
  toggle: PropTypes.shape({
    label: PropTypes.string,
    iconShape: PropTypes.string,
  }),
  onClick: PropTypes.func,
  level: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  children: PropTypes.node,
};

Accordion2Item.defaultProps = {
  id: '',
  toggle: {},
  onClick: null,
  level: null,
  children: null,
};

export default Accordion2Item;
