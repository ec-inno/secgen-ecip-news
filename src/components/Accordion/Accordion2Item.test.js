import React from 'react';
import renderer from 'react-test-renderer';
import Accordion2Item from './Accordion2Item';

describe('Accordion2Item', () => {
  const props = {
    id: 'accordion-item-test',
    children: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Accordion2Item {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
