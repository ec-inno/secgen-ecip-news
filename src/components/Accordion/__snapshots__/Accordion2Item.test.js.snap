// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion2Item renders correctly 1`] = `
<div
  className="ecl-accordion2__item"
>
  <div
    className="ecl-accordion2__content"
    hidden={true}
    id="accordion-item-test-content"
    role="region"
  >
    test
  </div>
</div>
`;
